src/cfg.rs

```diff
@@ -0,0 +1,120 @@
+use pwd::Passwd;
+use serde::{Deserialize, Serialize};
+use std::env;
+use std::fs::File;
+use std::collections::BTreeMap;
+
+const CONFIG_ENV: &str = "DOORMAN_CONFIG";
+const DEFAULT_CONFIG: &str = "/etc/doorman.yml";
+
+#[derive(Deserialize, Debug)]
+pub struct Options {
+    pub user: String,
+    pub dosemu_container: String,
+    pub rundir: String,
+    pub sysops: Vec<String>,
+}
+
+fn default_door_nodes() -> i8 {
+    return 1;
+}
+
+#[derive(Deserialize, Debug)]
+pub struct Door {
+    pub path: String,
+    #[serde(default = "default_door_nodes")]
+    pub nodes: i8,
+    pub launch: String,
+    pub configure: Option<String>,
+    pub nightly: Option<String>,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct Config {
+    pub doorman: Options,
+    pub doors: BTreeMap<String, Door>,
+}
+
+#[derive(Serialize, Debug)]
+pub struct User {
+    pub uid: u32,
+    pub username: String,
+    pub display_name: String,
+    pub is_sysop: bool,
+}
+
+impl Config {
```

Comment:
It's usually preferable to put the `struct` and `impl` blocks right next to each other, unless you have a very specific reason not to.